katsdpimageutils/primary_beam_correction.py

```diff
@@ -12,10 +12,9 @@
 _cosine_taper = katbeam.jimbeam._cosine_taper
 
 # Mapping from 'BANDCODE' to katbeam model name
-# NOTE: S-band defaults to L-band model since katbeam has no S-band models
 BAND_MAP = {'L': 'MKAT-AA-L-JIM-2020',
             'UHF': 'MKAT-AA-UHF-JIM-2020',
-            'S': 'MKAT-AA-L-JIM-2020'}
+            'S': 'MKAT-AA-S-JIM-2020'}
```

Comment:
I realise that I should have noticed this before, but I see in you're `check_band` function you define the range of S-band as being 2-4 GHz, but for MeerKAT the S-band system actually is from 1.75 - 3.5 GHz. Could you change the bottom of the range to 1.75 GHz in the `check_band`, otherwise there is a danger that images made with the lowest sub-bands won't be processed with the correct beam.